### PR TITLE
Fix a regression from #791.

### PR DESCRIPTION
**Pull request by realityking**
This required a small addition to the JInput API that should be helpful for others as well.

### CODE REVIEW

File: libraries/joomla/input/input.php [hunk 1]
```diff
@@ -183,15 +183,21 @@ public function getArray(array $vars, $datasource = null)
...
+	 * @param   boolean $overwrite  If true and an existing key is found, the value is overwritten, otherwise it is ignored.
 	 *
 	 * @return  void
 	 *
 	 * @since   11.1
 	 */
-	public function set($name, $value)
+	public function set($name, $value, $overwrite = true)
 	{
+		if (!$overwrite && array_key_exists($name, $this->data))
```
Comment (LouisLandry) [inline libraries/joomla/input/input.php]: isset() is substantially faster than array_key_exists() ... and this function may get called a few times in an execution cycle.  Might consider changing it.

**Comment by LouisLandry** [comment]
Convention-wise we've always had ->def() for this behavior.  Not that I think that's necessarily better, but it is worth noting.

Also, it would be great to add a couple of small unit tests just to make sure the functionality doesn't get lost somewhere down the line.

**Comment by realityking** [comment]
Either works for me. I followed how JRequest worked but I'm open to change it if we agree on def() as a convention.

I'll whip up some unit tests when I'm back home.

**Comment by realityking** [comment]
Either works for me. I followed how JRequest worked but I'm open to change it if we agree on def() as a convention.

I'll whip up some unit tests when I'm back home.

**Comment by robschley** [comment]
I like def() better than having another parameter in set. For what it's worth.

**Pull request merged**